viewing_party/party.py

```diff
@@ -1,23 +1,153 @@
+
+    # return genre with highest count of movies
+    most_watched = max(genre_count, key=genre_count.get)
+    return most_watched
+
+    
+# get list of movies user has watched, but friends haven't.
+def get_unique_watched(user_data):
+    unique_movies = []
+
+    friends_watched = []
+    for friend in user_data["friends"]:
+        for movie in friend["watched"]:
+            if movie not in friends_watched:
+                friends_watched.append(movie)
 
-# -----------------------------------------
-# ------------- WAVE 2 --------------------
-# -----------------------------------------
+    for movie in user_data["watched"]:
+        if movie not in friends_watched:
+            unique_movies.append(movie)
 
+    return unique_movies
 
-# -----------------------------------------
-# ------------- WAVE 3 --------------------
-# -----------------------------------------
 
+def get_friends_unique_watched(user_data):
+    friends_movie_for_user = []
+
+    user_watched = set()
+    for movie in user_data["watched"]:
+        user_watched.add(movie["title"])
+
+    friends_watched = set()
+    for friend in user_data["friends"]:
+        for movie in friend["watched"]:
+            if movie["title"] not in user_watched and movie["title"] not in friends_watched:
+                friends_movie_for_user.append(movie)
+                friends_watched.add(movie["title"])
+
+    return friends_movie_for_user
+        
         
-# -----------------------------------------
-# ------------- WAVE 4 --------------------
-# -----------------------------------------
+def get_available_recs(user_data):
+    recommended_list = []
+    subscriptions = set(user_data.get("subscriptions", []))
+    user_watched_titles = {movie["title"] for movie in user_data.get("watched", [])}
+    recommended_titles = set()
+
+    for friend in user_data.get("friends", []):
+        for movie in friend.get("watched", []):
+            if (
+                movie["title"] not in user_watched_titles
+                and movie["host"] in subscriptions
+                and movie["title"] not in recommended_titles
+            ):
+                recommended_list.append(movie)
+                recommended_titles.add(movie["title"])
+
+    return recommended_list
+
+def get_new_rec_by_genre(user_data):
+    recommendations = []
+
+    # Get most watched genre
+    most_genre = get_most_watched_genre(user_data)
+
+    if most_genre is None:
+        return recommendations
+
+    user_watched_titles = {movie["title"] for movie in user_data["watched"]}
+
+    # checking to see what movies friends have watched
+    for friend in user_data["friends"]:
+        for movie in friend["watched"]:
+            if (
+                movie["title"] not in user_watched_titles
+                and movie["genre"] == most_genre
+                and movie not in recommendations
+            ):
+                recommendations.append(movie)
+
+    return recommendations

```

Comment:
Similar to feedback on the prior function, we want movies the user has not seen but they're friends have, just like we get from `get_friends_unique_watched`, but now we need to layer on a filter to check if the movies also match the user's most watched genre. 

The current implementation is duplicating effort from `get_friends_unique_watched`, so how can we reuse our prior function to avoid repeating work?